Fortran/gfortran/utils/update-test-config.py

```diff
@@ -0,0 +1,633 @@
+# object to the MOUT list and return True. Otherwise, leave the MOUT list
+# unchanged and return False.
+def try_match(regex: re.Pattern, line: str, mout: list[re.Match[str]]) -> bool:
+    m = regex.search(line)
+    if m:
+        mout.append(m)
+        return True
+    return False
+
+# Count the number of elements in the list that match satisfy the predicate.
+def count_if(l, predicate):
+    return sum(1 for e in l if predicate(e))
+
+# Print a message. This is only around to save a bit of typing.
+def printf(fmt: str, *args) -> None:
+    print(fmt.format(*args))
+
+# Print a message in verbose mode.
+def message(fmt: str, *args) -> None:
+    printf(fmt, *args)
+
+# Print a warning message.
+def warning(fmt: str, *args) -> None:
+    printf('WARNING: ' + fmt, *args)
+
+# Print an error message and exit.
+def error(fmt: str, *args) -> None:
+    printf('ERROR: ' + fmt, *args)
+    exit(1)
+
+# The target is usually a regular expression. But the regex syntax used by
+# DejaGNU is not exactly the same as that supported by cmake. This translates
+# the DejaGNU regex to a cmake-compatible regex.
+def convert_target_regex(t: str) -> str:
+    # XXX: This translation is not strictly correct.
+    # In DejaGNU, the ? character matches a single character unless it follows
+    # an atom. In the target specifications in the gfortran test suite, this is
+    # only used as a single character match.
+    t = t.replace('?', '.')
+
+    # XXX: This translation is not strictly correct.
+    # in DejaGNU, the * character can also be a wildcard match for zero or more
+    # characters unless it follows an atom. In the target specifications in the
+    # gfortran test suite, it is only used as a wildcard.
+    t = t.replace('*', '.+')
+
+    return t
+
+# Parse the enabled targets from a target specification string. Some of the
+# targets may require additional compiler/linker options. Those options are
+# returned as well.
+def parse_enabled_targets(t: str) -> tuple[list[str], list[str]]:
+    targets = []
+    options = []
+
+    # An expression can be wrapped with braces. While this seems to be necessary
+    # for complex expressions, it can be used with simple expressions as well.
+    if t.startswith('{') and t.endswith('}'):
+        t = t[1:-1].strip()
+
+    # A simple expression may be a sequence of targets.
+    for tgt in t.split(' '):
+        if re_platform.match(tgt):
+            targets.append(convert_target_regex(tgt))
+        elif tgt in platforms:
+            targets.append(convert_target_regex(platforms[tgt]))
+        # Some "targets" need to be translated to compiler/linker flags.
+        elif tgt in ['fopenmp', 'fopenacc', 'pthread']:
+            options.append('-' + tgt)
+        elif tgt in ['c99_runtime']:
+            options.append('-lc')
+        elif tgt in [
+            'fd_truncate',
+            'fortran_large_int',
+            'fortran_real_10',
```

Comment:
I think `real(10)` is supported on x86 only. But it should be ok to ignore this for now.

Reply:
Thanks. I didn't know that. I might need some help when I get around to fixing that.